src/core/execute.c

```diff
@@ -2672,6 +2722,12 @@ static int exec_child(
                 return r;
         }
 
+        r = setup_keyring(unit, params, uid, gid);
+        if (r < 0) {
+                *exit_status = EXIT_KEYRING;
+                return r;
+        }
```

Comment:
I'm reading https://github.com/torvalds/linux/blob/4693fc734d675c5518ea9bd4c9623db45bc37402/Documentation/security/keys.txt#L253
> The value of the file /proc/self/attr/keycreate influences the labeling of
newly-created keys.  If the contents of that file correspond to an SELinux
security context, then the key will be assigned that context.  Otherwise, the
key will be assigned the current context of the task that invoked the key
creation request.

I.e. all keyrings will be labeled with the context of the `systemd`. Is this correct?